examples/datasets/generate_missing_values.py

```diff
@@ -0,0 +1,135 @@
+"""
+=============================================
+Generating MNAR / MCAR missing_values in data
+=============================================
+
+This example illustrates how the :class:`sklearn.datasets.ValueDropper` can
+be used to generate missing values completely at random or conforming to the
+given distribution.
+
+The :class`sklearn.datasets.ValueDropper` is a transformer which can be
+initialized with a ``missing_rate`` and a ``missing_distribution``. It
+provisions generating missing values incrementally (on top of the existing
+missing values) so that an exact fraction of missing values can be introduced
+for benchmarking missing-value handling strategies and evaluating the
+performance of such strategies with respect to the type and extent of
+missingness in data.
+
+MNAR or Missing Not At Random refers to the case when the missingness in the
+data is distributed not at random. It is either correlated with the target
+value(s) or with the data itself.
+
+MCAR or Missing Completely At Random refers to the case when the missingness
+in the data is completely random and does not correlate with the classification
+target value(s) or the data.
+"""
+# Author: Raghav RV <rvraghav93@gmail.com>
+#
+# License: BSD 3 clause
+
+from __future__ import print_function
+
+import numpy as np
+from sklearn.datasets import ValueDropper
+
+print(__doc__)
+
+
+X = np.array([[0, 1, 2],
+              [3, 4, 5],
+              [6, 7, 8],
+              [9, 0, 1],
+              [2, 3, 4],
+              [8, 9, 8],
+              [8, 9, 8],
+              [1, 0, 5],
+              [5, 4, 3],
+              [2, 1, 1],
+              [3, 4, 5],
+              [2, 3, 4],
+              [8, 9, 8],
+              [7, 8, 9]], dtype=float)
+y = np.array([1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 2, 2, 2, 2])
+
+# Drop 10% of values across all features for samples
+# where the target class label is randomly chosen
+
+vd = ValueDropper(missing_rate=0.1,
+                  missing_distribution={1: 1.},
```

Comment:
I vote to keep it simple and have just one missing_distribution parameter. It gives you all the degrees of freedom to specify missingness